models/index.js

```diff
@@ -1,81 +1,95 @@
-const User = require('./User');
-const Post = require('./Post');
-const Comment = require('./Comment');
-const Location = require('./Location');
-const Activity = require('./Activity');
-const Cuisine = require('./Cuisine');
-const PostLocation = require('./PostLocation');
-const PostActivity = require('./PostActivity');
-const PostCuisine = require('./PostCuisine');
-
-User.hasMany(Post, {
-    foreignKey: 'user_id'
-});
-
-Post.belongsTo(User, {
-    foreignKey: 'user_id',
-});
-
-User.hasMany(Comment, {
-    foreignKey: 'user_id',
-});
-
-Comment.belongsTo(User, {
-    foreignKey: 'user_id',
-});
-
-Post.hasMany(Comment, {
-    foreignKey: 'post_id'
-});
-
-Comment.belongsTo(Post, {
-    foreignKey: 'post_id',
-});
-
-Post.belongsToMany(Location, {
-    through: PostLocation,
-    as: 'locations',
-    foreignKey: 'post_id'
-});
-
-Location.belongsToMany(Post, {
-    through: PostLocation,
-    as: 'posts',
-    foreignKey: 'location_id'
-});
-
-Post.belongsToMany(Activity, {
-    through: PostActivity,
-    as: 'activities',
-    foreignKey: 'post_id'
-});
-
-Activity.belongsToMany(Post, {
-    through: PostActivity,
-    as: 'posts',
-    foreignKey: 'activity_id'
-});
-
-Post.belongsToMany(Cuisine, {
-    through: PostCuisine,
-    as: 'cuisine',
-    foreignKey: 'post_id'
-});
-
-Cuisine.belongsToMany(Post, {
-    through: PostCuisine,
-    as: 'posts',
-    foreignKey: 'cuisine_id'
-});
-
-module.exports = { 
-    User, 
-    Post, 
-    Comment, 
-    Location, 
-    Activity, 
-    Cuisine,
-    PostLocation,
-    PostActivity,
-    PostCuisine
-}
\ No newline at end of file
+const User = require('./User');
+const Post = require('./Post');
+const Comment = require('./Comment');
+const Location = require('./Location');
+const CountryCode = require('./CountryCode');
+const Activity = require('./Activity');
+const Cuisine = require('./Cuisine');
+const PostLocation = require('./PostLocation');
+const PostActivity = require('./PostActivity');
+const PostCuisine = require('./PostCuisine');
+
+User.hasMany(Post, {
+    foreignKey: 'user_id'
+});
+
+Post.belongsTo(User, {
+    foreignKey: 'user_id',
+});
+
+User.hasMany(Comment, {
+    foreignKey: 'user_id',
+});
+
+Comment.belongsTo(User, {
+    foreignKey: 'user_id',
+});
+
+Post.hasMany(Comment, {
+    foreignKey: 'post_id'
+});
+
+Comment.belongsTo(Post, {
+    foreignKey: 'post_id',
+});
+
+Post.belongsToMany(Location, {
+    through: PostLocation,
+    as: 'locations',
+    foreignKey: 'post_id'
+});
+
+Location.belongsToMany(Post, {
+    through: PostLocation,
+    as: 'posts',
+    foreignKey: 'location_id'
+});
+
+Post.belongsToMany(CountryCode, {
+    through: PostLocation,
```

Comment:
"PostLocation" is the through table created for the relationship between Post table and Location table, we can't reference that here. 

Does a separate through table need created for the relationship between Post and CountryCode? 
If so, need to create PostCountrycode through model that has post ID and country code ID as entries. 

If we wanted to make it easier on ourselves, we could limit each post to a single Country Code instead of doing a many-to-many relationship here.

Reply:
This has been resolved by creation of 'PostCountryCode' table and relationships kept similar to other through tables.